cosmovisor/process.go

```diff
@@ -46,6 +46,10 @@ func (l Launcher) Run(args []string, stdout, stderr io.Writer) (bool, error) {
 	if err := cmd.Start(); err != nil {
 		return false, fmt.Errorf("launching process %s %s failed: %w", bin, strings.Join(args, " "), err)
 	}
+	err = doPreUpgrade(l.cfg)
```

Comment:
It should run before the upgrade. I think having it here forces the binary to execute pre-upgrade everytime when the DAEMON restarts. Best place would be to call this just before `DoUpgrade` below